src/main/java/org/embulk/input/kintone/KintoneInputPlugin.java

```diff
@@ -74,14 +87,30 @@ public TaskReport run(TaskSource taskSource,
                 client.validateAuth(task);
                 client.connect(task);
 
-                CreateCursorResponseBody cursor = client.createCursor(task);
+                CreateCursorResponseBody cursor = client.createCursor(task, schema);
                 GetRecordsByCursorResponseBody cursorResponse = new GetRecordsByCursorResponseBody(true, null);
 
+                List<String> subTableFieldCodes = null;
+                if (task.getExpandSubtable()) {
+                    subTableFieldCodes = client.getFieldCodes(task, FieldType.SUBTABLE);
+                }
+
                 while (cursorResponse.isNext()) {
                     cursorResponse = client.getRecordsByCursor(cursor.getId());
                     for (Record record : cursorResponse.getRecords()) {
-                        schema.visitColumns(new KintoneInputColumnVisitor(new KintoneAccessor(record), pageBuilder, task));
-                        pageBuilder.addRecord();
+                        List<Record> records;
+                        if (task.getExpandSubtable()) {
+                            records = expandSubtable(record, subTableFieldCodes);
```

Comment:
NOTE:
If expand subtable, but not specify fields inside subtable, duplicate records will be generated because the subtable remains expanded.